pkgs/development/python-modules/Theano/compiler.patch

```diff
@@ -0,0 +1,16 @@
+--- Theano-src/theano/configdefaults.py_	2017-06-14 14:12:00.000000000 -0400
++++ Theano-src/theano/configdefaults.py	2017-06-14 14:12:44.000000000 -0400
+@@ -380,11 +380,11 @@
+             'FAST_COMPILE', 'PROFILE_MODE', 'DEBUG_MODE'),
+     in_c_key=False)
+ 
+-param = "g++"
++param = "@g++@"
+ 
+ # Test whether or not g++ is present: disable C code if it is not.
+ try:
+-    rc = call_subprocess_Popen(['g++', '-v'])
++    rc = call_subprocess_Popen(['@g++@', '-v'])
```

Comment:
Directly using `g++` is indeed not going to work because with Nix we wrap our compilers as you've noticed. So its likely best here to just use `stdenv.cc` (though that might not be ideal on Darwin since they use LLVM).